frontend/src/components/dashboard/settings/Settings.tsx

```diff
@@ -56,12 +70,23 @@ const Settings = () => {
     )
   }
 
+  function onAddCustomDomainClicked() {
+    modalContext.setModalContent(
+      <AddCustomDomainModal onSuccess={fetchCustomFromAddresses} />
+    )
+  }
+
   async function fetchUserSettings() {
     const { hasApiKey, creds } = await getUserSettings()
```

Comment:
`GET /settings` currently returns info about the API key as well as a list of credentials. Will it make sense to include the custom From Addresses as part of the response from this call?

Reply:
Is there a particular advantage in doing so, besides saving one network call?  the endpoint GET /email/from has to exist, for the dropdown in EmailTemplate.  IMO it doesn't fit into getUserSettings because the email_from_address table does not associate the email address to a user. 

Alternatively, if you want to centralize the retrieval of credentials / from addresses, then maybe we should also get rid of GET /telegram/credentials and GET/sms/credentials

Reply:
Agree on no additional benefit beside saving the one API call. `GET /telegram/credentials` and `GET /sms/credentials` also can't be removed since they're used in the creation flow too. There's also no good way to link between the user and From Addresses he's allowed to access. 

I think we can keep this as it is.